src/views/user/ResetPassword/ResetPassword.jsx

```diff
@@ -0,0 +1,307 @@
+    if (!input.value) {
+      this.validity.passwordIsValid = false;
+      return;
+    }
+    const regex = new RegExp(['^(?=.*[\\d])(?=.*[A-Z])(?=.*[a-z])',
+      '(?=.*[!@#$%^&*])[\\w!@#$%^&*]{8,}$'].join(''));
+    const { confirm } = this.state;
+    if (!regex.test(input.value)) {
+      this.validity.passwordIsValid = false;
+      this.validity.passwordIsConfirmed = confirm === input.value;
+      return;
+    }
+    this.validity.passwordIsValid = true;
+    this.validity.passwordIsConfirmed = confirm === input.value;
+  }
+
+  /**
+   * @param {Object} errorResonse http error status code
+   * @returns {String} error response message
+   */
+  serverErrorResponse = (errorResonse) => {
+    const { statusCode, message } = errorResonse;
+    switch (statusCode) {
+    case 400:
+      return 'The password reset token is empty';
+
+    case 401:
+      return 'The reset token has expired or is invalid';
+
+    case 404:
+      return 'No user was found';
+
+    case 409:
+      return 'Password does not match';
+
+    default:
+      return message.includes('Network Error')
+        ? 'There was a connection problem.'
+        : '0ops! Something went wrong. Refresh and try again';
+    }
+  }
+
+  /**
+   * handle form submission
+   * @param {Object} event event object
+   * @returns {null} null
+   */
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const {
+      passwordIsValid,
+      passwordIsConfirmed
+    } = this.validity;
+
+    if (!passwordIsValid || !passwordIsConfirmed) return;
+    const { password } = this.state;
+
+    const { token } = this.props;
+    localStorage.setItem('token', token);
+    this.setState({ password: '', confirm: '' });
+    updatePassword('/users/reset-password', password);
+  }
+
+  /**
+   * handle change on form fields
+   * @param {Object} event event object
+   * @returns {null} null
+   */
+  handleInput = (event) => {
+    const input = event.target;
+
+    this.validatePassword(input);
+    this.validateConfirmPassword(input);
+
+    const {
+      passwordIsValid,
+      passwordIsConfirmed,
+    } = this.validity;
+    let disable = true;
+
+    const isEmptyPasswordInput = input.name === 'password'
+    && !input.value;
+    const isEmptyConfirmInput = input.name === 'confirm'
+    && !input.value;
+
+    if (passwordIsValid || isEmptyPasswordInput) {
+      this.frontendValidations.password.length = 0;
+    }
+
+    if (passwordIsConfirmed || isEmptyConfirmInput) {
+      this.frontendValidations.confirm.length = 0;
+    }
+
+    if (passwordIsValid && passwordIsConfirmed) disable = false;
+
+    this.setState({
+      [event.target.name]: event.target.value,
+      disabled: disable,
+      frontendValidations: this.frontendValidations,
+    });
+  }
+
+  /**
+   * handle password visibility
+   * @param {Object} event event object
+   * @returns {undefined} undefined
+   */
+  handlePasswordVisibility = (event) => {
+    const el = event.target;
```

Comment:
what is `el` ?

Reply:
It's the target element. The element that triggered the event. I have renamed it to `target`.